GetStream/stream-chat-swift
# Background Mapping - Offline Fixes

### 🔗 Issue Links

Continuation of https://github.com/GetStream/stream-chat-swift/pull/2831

### 🎯 Goal

The goal of this PR was to fix some deficiencies BG Mapping had when being Offline

### 📝 Summary

There was an assumption that for the initial fetch we would not need to notify the listeners. This was not a problem when being online, given that any interaction with the network would trigger updates, and therefore the listeners would be notified.

This is, though, a big problem when in offline mode, where we depend on the initial data.

### 🛠 Implementation

This PR makes sure the listener is always notified with the initial fetch

### 🧪 Manual Testing Notes

1. Open the app
2. Close the app
3. Disable internet
4. Open the app while offline (should have some content fetched in another session)

Expected result:
- The data in the channel list should be displayed
- Opening a channel results in a message list filled with the messages fetched in a previous session

### ☑️ Contributor Checklist


### 🎁 Meme

![](https://media.giphy.com/media/avOd4dltPp9fIDm5P3/giphy.gif)
